src/components/QueryBuilder.vue

```diff
@@ -57,42 +58,21 @@ export default Vue.extend( {
 		};
 	},
 	methods: {
-		validate(): boolean {
-			this.errors = [];
-			this.$store.dispatch( 'setErrors', [] );
-			if ( !this.selectedProperty && !this.textInputValue ) {
-				this.errors.push( {
-					// eslint-disable-next-line max-len
-					message: 'Looks like the Query Builder was empty, please enter a valid query first, then try running it again',
-					type: 'notice',
-				} );
-				return false;
-			}
-
-			if ( !this.selectedProperty || !this.textInputValue ) {
-				if ( !this.selectedProperty ) {
-					this.fieldErrors.property = {
-						message: 'Please select a property',
-						type: 'error',
-					};
-				}
-				if ( !this.textInputValue ) {
-					this.fieldErrors.value = {
-						message: 'Please enter a value',
-						type: 'error',
-					};
-				}
-				this.errors.push( {
-					message: 'One or more fields are empty. Please complete the query or select a fitting field type.',
-					type: 'error',
-				} );
-				return false;
-			}
-			return true;
+		validate(): ValidationResult {
+			const formValues = {
+				property: this.selectedProperty,
+				value: this.textInputValue,
+			};
+			const validator = new Validator( formValues );
+			const validationResult = validator.validate();
+			this.errors = validationResult.formErrors;
+			this.$store.dispatch( 'setErrors', validationResult.formErrors );
+			this.fieldErrors = validationResult.fieldErrors;
+			return validationResult;
```

Comment:
It should not change the state _and_ return something. This violates command-query-separation. We can do without returning the result here and just check `this.errors` in the if-condition below.

Reply:
You're absolutely right. Done now.